interview-questions.md

```diff
@@ -7,65 +7,93 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. What does it mean if an application has CRUD? What does CRUD stand for?
 
-  Your answer:
+  Your answer: CRUD stands for create, read, update, delete. If an application has this it means that it is considered to have full functionality. Essentially, the model can has functions that create, display, update, and delete content.
 
-  Researched answer:
+  Researched answer: After looking through the internet I found that CRUD makes up the four basic components of persistent storage for a web app. 
 
 
 
 3. What is a migration? Why would you use one?
 
-  Your answer:
+  Your answer: A migration is a file that is used to create, update, or deletes parts of a model. 
 
-  Researched answer:
+  Researched answer: After loking in the syllabus 
+  it looks like migrations are used to modify the shape of databases by adding or deleting columns, changing the name of the column, or changing the data type of a column.
 
 
 
 4. Describe as many possible relationships between a model called Person and a model called Computer as you can think of. Based on each description, which model would hold the foreign key?
 
   Your answer:
 
-  Researched answer:
+   class Person 
+   has_many :computers 
+
+   class Computer 
+   belongs_to :person >>> Holds the foreign key
+   ------------------------------------------------------------
+
+   class Computer 
+   has_many :persons 
+
+   class Person 
+   belongs_to :computer >>> Holds the foreign key
+   -------------------------------------------------------------
+
+   class Person
+   has_one :computers 
+
+   class computers
+   belongs_to :person >>> Holds the foreign key
+   -------------------------------------------------------------
+
+
+  Researched answer: After researching it looks like my answer is mostly correct but maybe I would add a many_to_many relationship as well
 
 
 
 5. What is object-relational mapping?
 
-  Your answer:
+  Your answer: ORM is a programming technique that allows information from two or more different languages to be transfered and used
 
-  Researched answer:
+  Researched answer: After looking at the syllabus it looks like I was a little off in that ORM allows incompatible systems to convert data by using object-oriented programming languages.
 
 
 
 6. What is a gem?
 
-  Your answer:
+  Your answer: A gem is downloadable extension of Ruby
 
-  Researched answer:
+  Researched answer: After looking into this I found that even Rails is considered a gem
 
 
 
 7. Why is it important to have validations in your application?
 
-  Your answer:
+  Your answer: Validations ensure that only desired data enters the database of an app.
```

Comment:
Again, I think I'd want to elaborate a little further, explaining how we need to keep the database clean and provide some basic securities.